### PR TITLE
Refine chat response.

### PR DESCRIPTION
**Pull request by mathewjordan**
- Implement Chat-based queries.
- Shift Chat to search screen, rename directory.
- Partially remove localstorage dependencies in chat component.  Pull question data from primary Search bar
- More work
- Make straming answers work more reliably.
- Incomplete search pieces to help with generative ai search
- Wire up not logged in modal on Search page directing users to login if they want to use generative AI search
- Refine chat response.

### CODE REVIEW

File: hooks/useMarkdown.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      const updatedHtml = hasCursor
+        ? htmlContent.replace(
+            cursorRegex,
+            `<span class="markdown-cursor"></span>`
+          )
+        : htmlContent;
+
+      setHtml(updatedHtml);
+    })();
+  }, [preparedMarkdown]);
```
Comment (adamjarling) [inline hooks/useMarkdown.tsx:45]: Do we need `hasCursor` in the dependency array as well?

File: hooks/useMarkdown.tsx
Comment (adamjarling) [inline hooks/useMarkdown.tsx:1]: This hook is fantastic.  👍

File: hooks/useChatSocket.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ const useChatSocket = () => {
   const [authToken, setAuthToken] = useState<string | null>(null);
 
   useEffect(() => {
+    console.log(DCAPI_CHAT_ENDPOINT);
```
Comment (adamjarling) [inline hooks/useChatSocket.ts:11]:
```suggestion
```

File: components/Shared/BouncingLoader.tsx
Comment (adamjarling) [inline components/Shared/BouncingLoader.tsx:1]: The loader looks awesome too.

**Review by adamjarling — COMMENTED** [review]
This is awesome, and can't wait to see it rebased on latest `preview/chat-integration`.

I know it's nit-picky, but when we're in files, if possible let's lean on the import path aliases`@` so when we move stuff around it's less hassle.

```tsx
import { StyledStreamedAnswer } from "./Response.styled";

// prefer this
import useMarkdown from "@/hooks/useMarkdown";
```

**Comment by mathewjordan** [comment]
Thanks for taking a look @adamjarling. This is ready for another review.

**Review by adamjarling — APPROVED** [review]
Looking great.  Let's get this in place.

**Pull request merged**